#162949043 Pagination for articles

Paulvitalis200 opened the pull request:
#### What does this PR do?

Enables the articles page to have pagination.

#### Description of Task to be completed?
- Create a method for getting active page
- Use Semantic React pagination to do pagination

#### How should this be manually tested?
1. Navigate to a folder where you would prefer to test the repository.

2. Create a directory
 ```$```  ```mkdir myproject```

3. Navigate to the directory "myproject"
 ```$```   ```cd myproject```
4. Clone the author's haven repository
  ```$```   ```git clone https://github.com/andela/ah-robotics-frontend.git```
5. Change the directory into the "ah-robotics-fronted"
  ```$```   ```cd ah-robotics-frontend```

6.  Checkout to the branch ft-pagination-162949043
 ```$```   ```git checkout ft-pagination-162949043```

7. Install dependancies from the package.json
 ```$```   ```npm install``` to install all the modules listed in the `package.json` file.

8. Start the application in development mode
  ```$```   ```npm start``` to run the app in the development mode 

The application opens http://localhost:3000 in the browser.
Change the URL on the browser to;
 ```
https://ah-robotics-frontend-sta-pr-18.herokuapp.com
```

9. Scroll down to the bottom of the page to access the pagination links. Click on a specific page to change the page being viewed.

###  Running Tests
```
npm test
```

#### What are the relevant pivotal tracker stories?
[#162949043](https://www.pivotaltracker.com/n/projects/2235171/stories/162949043)

### Screenshot
Image showing the pagination
![image](https://user-images.githubusercontent.com/10106044/53358160-990d8100-3940-11e9-8b42-2fe94fa6f0b2.png)



#### Questions:

johnwayodi committed:
feat(userRegistration): add registration functionality

- create the registration UI
- add registration views, actions, reducers, saga

[Starts #162949035]

johnwayodi committed:
feat(userRegistration): add registration functionality

- create the registration UI
- add registration views, actions, reducers, saga

[Starts #162949035]

johnwayodi committed:
feat(register): add email verification functionality

- add email verification view, reducers, component

[#162949035]

johnwayodi committed:
feat(register): add more tests

- add tests for register and verification functionality

[#162949035]

johnwayodi committed:
feat(register): refactor code

- refactor code to solve codacy issues

[#162949035]

jomasim committed:
ft(aricles) : Creating,fetching single and fetching all articles

- creating article listing page
- creating article editing page
- set up article actions
- add articles saga and reducer

[Starts #162949040]

jomasim committed:
ft(articles): posting article
- designing article posting page
- add create article saga
- add create reducer
- add create article action creators

[#162949040]

jomasim committed:
ft(articles): add tests

- add post article reducer test
- refactor code

[#162949040]

Paulvitalis200 committed:
feat(pagination): get all articles

- get articles

[Starts #162949043]

Paulvitalis200 committed:
feat(pagination): add pagination functionality

- install react-js-pagination
- add pagination to articles page
- add handlePageChange function

[#162949043]

Paulvitalis200 committed:
feat(pagination): update pagination

- use semantic react pagination component

[Finishes #162949043]

Paulvitalis200 committed:
feat(pagination): fix login bug

- update url routes

[#162949043]

Paulvitalis200 committed:
feat(pagination): fix merge conflicts

- fix merge conflicts with develop

[#162949043]

Paulvitalis200 committed:
feat(pagination): refactor code

- refactor code
- fix conflicts

[#162949043]

sharkdevs commented on src/App.js:
```diff
@@ -17,14 +18,16 @@ class App extends Component {
             <Route path="/resetpassword" component={ResetPasswordView} />
             <Route path="/reset_password/:token" component={NewPasswordView} />
+            <Route path="/" component={ArticlesView} exact />
+            <Route path="/login" component={LoginView} exact />
```
This Login route is a duplicate of the one above.

Paulvitalis200 replied:
Thanks for the observation. I have fixed this

sharkdevs reviewed (commented):
The tests are passing fine and I was able to seamlessly test the pagination. However, consider the change below. Regards.

Paulvitalis200 committed:
feat(pagination): add tests

- add pagination test

[#162949043]

Paulvitalis200 committed:
feat(pagination): refactor code

- refactor fetchArticleSaga
- refactor article actions
- refactor Article listing view

[#162949043]

Teatoller reviewed (approved):
@Paulstar200 ,
I have followed the instructions you have enumerated with ease and managed to experience the feature you put up working brilliantly.
The tests are also passing.

Regards
Steven Ennis

johnwayodi reviewed (commented):
nice work @Paulstar200  It is nice seeing that the articles are now paginated.

waracci reviewed (commented):
I have been able to click through the pages and the links work as expected.

Paulvitalis200 committed:
feat(pagination): style articles

- update article font family
- update article font size
- update article color

[#162949043]
